src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -831,4 +850,162 @@ private static void SetGlobalSettings(
+        ParseBooleanSetting(configuration, "mapNotRunnableToFailed", logger, result => settings.MapNotRunnableToFailed = result);
+        ParseBooleanSetting(configuration, "treatDiscoveryWarningsAsErrors", logger, result => settings.TreatDiscoveryWarningsAsErrors = result);
+        ParseBooleanSetting(configuration, "considerEmptyDataSourceAsInconclusive", logger, result => settings.ConsiderEmptyDataSourceAsInconclusive = result);
+        ParseBooleanSetting(configuration, "treatClassAndAssemblyCleanupWarningsAsErrors", logger, result => settings.TreatClassAndAssemblyCleanupWarningsAsErrors = result);
+        ParseBooleanSetting(configuration, "considerFixturesAsSpecialTests", logger, result => settings.ConsiderFixturesAsSpecialTests = result);
+        ParseBooleanSetting(configuration, "cooperativeCancellationTimeout", logger, result => settings.CooperativeCancellationTimeout = result);
+        ParseBooleanSetting(configuration, "orderTestsByNameInClass", logger, result => settings.OrderTestsByNameInClass = result);
+        ParseBooleanSetting(configuration, "parallelism:enabled", logger, result => settings.OrderTestsByNameInClass = result);
+
+        ParseIntegerSetting(configuration, "timeout:test", logger, result => settings.TestTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:assemblyCleanup", logger, result => settings.AssemblyCleanupTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:assemblyInitialize", logger, result => settings.AssemblyInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:classInitialize", logger, result => settings.ClassInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:classCleanup", logger, result => settings.ClassCleanupTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:testInitialize", logger, result => settings.TestInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:testCleanup", logger, result => settings.TestCleanupTimeout = result);
+
+        ParseStringSetting(configuration, "settingsFile", logger, fileName => settings.TestSettingsFile = fileName);
+
+        if (configuration["mstest:classCleanupLifecycle"] is string classCleanupLifecycle)
+        {
+            if (TryParseEnum(classCleanupLifecycle, out ClassCleanupBehavior lifecycle))
+            {
+                settings.ClassCleanupLifecycle = lifecycle;
+            }
+            else
+            {
+                throw new AdapterSettingsException(string.Format(
+                    CultureInfo.CurrentCulture,
+                    Resource.InvalidClassCleanupLifecycleValue,
+                    classCleanupLifecycle,
+#if NET
+                    string.Join(", ", Enum.GetNames<ClassCleanupBehavior>())));
+#else
+                    string.Join(", ", EnumPolyfill.GetNames<ClassCleanupBehavior>())));
```

Comment:
Can't we use polyfill for both? @Evangelink how's the story of polyfill here?

Reply:
I would need to double check but yeah most likely it should work.